### DISCUSSION TITLE
Spine 1.2.x: How to get the .dll files or where is the source to compile to get the .dll

### DISCUSSION THREAD

**Discussion by MSS970** [Cacti/cacti#5244] [Q&A]
Hi,
Apologies if this discussion thread does not below to Spine, but I wasn't able to raise a query in the Cacti/Spine as the discussion is not available.

I've just downloaded the latest code of Spine 1.2.x and compile it in Windows 2019 server, spine.exe file has been already generated.
However, I haven't seen the required .dll files.

Could you please advise where would I download them from?
or where is the source code of these file to compile and generate the .dll?

Thanks team in advance.

Much appreciated.

**Comment by TheWitness** [comment]
You are best to simply install CYGWIN and the required libraries.  You will need both MySQL and Net-SNMP develop packages and the respective libraries.  Once all that is installed on top of windows, the spine binary should work and the source should compile.

**Reply by MSS970** [reply]
I already have CYGWIN and the spine.exe has been compiled successfully.
My query is about the cyg*.dll files: is there any update on them?
below is a screenshot of the files in question.

![cyg-------dll-files](https://user-images.githubusercontent.com/124117691/220157380-34cfea62-47fb-4be2-bb13-cf5827b53320.PNG)

**Comment by netniV** [comment]
Alternatively, you can use the windows installer we have to install all the pre-requisites for runtimes, then just copy over your later version of spine if we don't have the current one done.

**Reply by MSS970** [reply]
I though about this idea, but since the cyg*.dll are Windows based dynamic libraries, there might be a prerequisite for their versions.
However, I have already copied the newly compiled spine.exe 1.2.24 (using CYGWIN) on top of the spine.exe which was installed from spine 1.2.23.

I am monitoring for now and will keep you gentlemen posted.

**Comment by MSS970** [comment]
It did not work:

20-Feb-2023 20:30:07 - SYSTEM STATS: Time:6.6626 Method:spine Processes:2 Threads:2 Hosts:48 HostsPerProcess:24 DataSources:386 RRDsProcessed:201
2023-02-20 20:30:03 - SPINE: Poller[Main Poller] PID[946] PT[42950184016] Device[Mi-STACK] Hostname[10.1.1.3] ERROR: HOST EVENT: Device is DOWN Message: Device responded to SNMP, ICMP: Ping timed out
2023-02-20 20:30:02 - SPINE: Poller[Main Poller] PID[947] PT[42950190208] Device[CUCM-SUB-Mi] Hostname[192.168.11.30] ERROR: HOST EVENT: Device is DOWN Message: ICMP: Ping timed out
2023-02-20 20:30:02 - SPINE: Poller[Main Poller] PID[947] PT[42950188560] Device[Cisco Unity Connection] Hostname[192.168.1.105] NOTICE: HOST EVENT: Device Returned from DOWN State
20-Feb-2023 20:25:06 - SYSTEM STATS: Time:5.6624 Method:spine Processes:2 Threads:2 Hosts:48 HostsPerProcess:24 DataSources:386 RRDsProcessed:220
20-Feb-2023 20:20:06 - SYSTEM STATS: Time:5.6531 Method:spine Processes:2 Threads:2 Hosts:48 HostsPerProcess:24 DataSources:386 RRDsProcessed:204
2023-02-20 20:20:02 - SPINE: Poller[Main Poller] PID[1462] PT[42950190032] Device[Mi-STACK] Hostname[10.1.1.3] NOTICE: HOST EVENT: Device Returned from DOWN State
2023-02-20 20:20:02 - SPINE: Poller[Main Poller] PID[1464] PT[42950181088] Device[Cisco Unity Connection] Hostname[192.168.1.105] ERROR: HOST EVENT: Device is DOWN Message: ICMP: Ping timed out
20-Feb-2023 20:15:06 - SYSTEM STATS: Time:5.6770 Method:spine Processes:2 Threads:2 Hosts:48 HostsPerProcess:24 DataSources:386 RRDsProcessed:226

**Comment by TheWitness** [comment]
To do ICMP ping, you have to set the spine binary setuid().  I think you would need the full install of Cygwin to get that as it requires RAW sockets.
